Format function for filename

benjamin-ross opened the pull request:
- Can format filename or not(will use the default filename in that case).
- Move 'fs-extra' to dependencies in order to use the generator outside.

Commit:
feat(Generator): Can format filename or not.

louis-bompart commented on src/generator/sfdx-generator.ts:
```diff
@@ -103,7 +110,10 @@ export class Generator {
 
     fs.writeFileSync(
-      path.resolve(this.options.outputDirectory, "generatedClient." + this.options.fileExtension),
+      path.resolve(
```
Try using https://nodejs.org/api/path.html#path_path_format_pathobject I think that it'll be more appropriate.

louis-bompart commented on tools/generate-files.js:
```diff
@@ -1,9 +1,13 @@
-
 var root = path.resolve(__dirname, './..')
-var generator = new generator.Generator("sfdx", root)
+var generator = new generator.Generator({
```
Split the instanciation and the declaration of the default option, like:
```javascript
const defaultGeneratorOptions = {foo: 'bar'};
const generator = new generator.Generator(defaultGeneratorOptions);
 ```

louis-bompart replied:
In fact, add the defaultOptions into `sfdx-generator.ts` and set the constructor default value to it.

benjamin-ross replied:
Not sure about that one, it's only an example of how to use it. The implementation will be determined by the project that use it. I prefer enforce the user to provide the options.

louis-bompart replied:
Then, create an interface for the options and ensure that coveo/sfdx-js and coveo/sfdx4csharp are using it properly.

louis-bompart commented on tools/generate-files.js:
```diff
@@ -1,9 +1,13 @@
 var generator = require('../dist/sfdx-generator.cjs')
```
Use const whenever possible instead of var while at it. If not possible, use let. If still not then use var.
(TL;DR: const>let>var)

louis-bompart reviewed (changes requested):
Good stuff, some minor changes and it'll be awesome and good to be shipped 👍

Commit:
refactor(sfdx-generator.ts): Changes the path creation for generated files

BREAKING CHANGE: Uses the god practice of nodejs path extension formating, instead of providing as
file extension like 'ts', we must now provide '.ts'.

marcantoineveilleux commented:
:tada: This PR is included in version 2.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/sfdx-generator)
- [GitHub release](https://github.com/coveo/sfdx-generator/releases/tag/v2.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
